optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveSetTree.java

```diff
@@ -40,22 +39,20 @@
  * @param <Point_> The type of the point (examples: int, LocalDateTime)
  * @param <Difference_> The type of the difference (examples: int, Duration)
  */
-public class ConsecutiveSetTree<Value_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>> {
+public final class ConsecutiveSetTree<Value_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>>
+        implements ConsecutiveInfo<Value_, Difference_> {
     private final Function<Value_, Point_> indexFunction;
     private final BiFunction<Point_, Point_, Difference_> differenceFunction;
     private final BiFunction<Difference_, Difference_, Difference_> sumFunction;
     private final Difference_ maxDifference;
     private final Difference_ zeroDifference;
-    private final Map<Value_, Integer> itemToCountMap;
-    private final NavigableSet<Value_> itemSet;
+    private final NavigableMap<Value_, Integer> itemToCountMap;
     private final NavigableMap<Value_, SequenceImpl<Value_, Difference_>> startItemToSequence;
     private final NavigableMap<Value_, BreakImpl<Value_, Difference_>> startItemToPreviousBreak;
 
     private final MapValuesIterable<Value_, SequenceImpl<Value_, Difference_>> sequenceList;
     private final MapValuesIterable<Value_, BreakImpl<Value_, Difference_>> breakList;
 
-    private final ConsecutiveDataImpl<Value_, Difference_> consecutiveData;
```

Comment:
Instead of copying the data to a new type, I made this type implement the same interface.